src/utils/crypto.js

```diff
@@ -1,16 +1,26 @@
 /* eslint-disable */
 import * as bitcoin from "bitcoinjs-lib";
 import * as ecc from "tiny-secp256k1";
-import { TESTNET, ASSUMED_TX_BYTES, BITCOIN_PRICE_API_URL, DEFAULT_DERIV_PATH } from "@lib/constants";
-import BIP32Factory from "bip32";
-import { ethers } from "ethers";
+import { TESTNET, ASSUMED_TX_BYTES, BITCOIN_PRICE_API_URL, FEE_LEVEL, MEMPOOL_API_URL } from "@lib/constants.config";
+
 import { ECPairFactory } from "ecpair";
 
 bitcoin.initEccLib(ecc);
-const bip32 = BIP32Factory(ecc);
 const ECPair = ECPairFactory(ecc);
 
-export const outputValue = (currentUtxo, sendFeeRate) => currentUtxo.value - sendFeeRate * ASSUMED_TX_BYTES;
+export const outputValue = (currentUtxo, sendFeeRate, price) =>
+    price || currentUtxo.value - sendFeeRate * ASSUMED_TX_BYTES;
+
+export function calculateFee({ vins, vouts, recommendedFeeRate, includeChangeOutput = true }) {
+    const baseTxSize = 10;
+    const inSize = 180;
+    const outSize = 34;
+
+    const txSize = baseTxSize + vins * inSize + vouts * outSize + includeChangeOutput * outSize;
+    const fee = txSize * recommendedFeeRate;
+
```

Comment:
TODO in future: calculate exact fees based on address types:

P2PKH (Pay-to-PubkeyHash):
Input size: ~148 virtual bytes (vB)
Output size: ~34 vB

P2SH (Pay-to-ScriptHash):
Input size: Variable, depending on the redeem script (e.g., ~295 vB for a typical 2-of-3 multisig)
Output size: ~32 vB

P2WPKH (Pay-to-Witness-Public-Key-Hash):
Input size: ~68 vB
Output size: ~31 vB

P2TR (Pay-to-Taproot):
Input size: ~57.5 vB (single key spend), variable for more complex scripts using Tapscript
Output size: ~43 vB